### PR TITLE
Add video recorder tutorial

### PR DESCRIPTION
**Pull request by iche033**
Add video recorder tutorial with description of the sim time and lock step features added in pull request #414 and #419

### CODE REVIEW

File: tutorials/video_recorder.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+\page videorecorder Video Recorder
+
+## Using the video recorder plugin
+
+Ignition Gazebo offers an video recorder tool for recording videos from the 3D
```
Comment (chapulina) [inline tutorials/video_recorder.md:5]:
```suggestion
Ignition Gazebo offers a video recorder tool for recording videos from the 3D
```

File: tutorials/video_recorder.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+that already has this plugin included in the GUI.
+
+```
+ign gazebo -v 4 video_record_dbl_pendulum.sdf
+```
+
+In this plugin, you should see a single button with a video recorder icon.
+Clicking on the button gives you the video format options that are available.
+
+![Video Recorder formats](https://github.com/ignitionrobotics/ign-gazebo/raw/7dccc0a08f9e926b9460c7662d49dbb0ce24ea61/tutorials/files/video_recorder/video_recorder.png)
```
Comment (chapulina) [inline tutorials/video_recorder.md:19]: I recently learned that we can refer to images using `@image html` like this:

https://github.com/ignitionrobotics/ign-gazebo/blame/ign-gazebo4/tutorials/gui_config.md#L38

File: tutorials/video_recorder.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+Once an option is selected, recording starts immediately and indicated by
```
Comment (chapulina) [inline tutorials/video_recorder.md:21]: Maybe this?

```suggestion
Once an option is selected, recording starts immediately as indicated by
```

**Review by chapulina — APPROVED** [review]
LGTM, just some minor comments.

File: tutorials/video_recorder.md (hunk 3 above)
Comment (iche033) [reply tutorials/video_recorder.md]: fixed 46a7d1f

File: tutorials/video_recorder.md (hunk 1 above)
Comment (iche033) [reply tutorials/video_recorder.md]: fixed. 46a7d1f

File: tutorials/video_recorder.md (hunk 2 above)
Comment (iche033) [reply tutorials/video_recorder.md]: updated. 46a7d1f

Is there a way to preview it? it does not seem to work locally and on github though.

Comment (chapulina) [reply tutorials/video_recorder.md]: Yup you can `make doc` and see the doxygen-generated tutorials on a browser

Comment (iche033) [reply tutorials/video_recorder.md]: ah I just had to clear the build cache and rebuild and now it's showing up locally.

File: tutorials/video_recorder.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
+\page videorecorder Video Recorder
+
+## Using the video recorder plugin
+
+Ignition Gazebo offers a video recorder tool for recording videos from the 3D
+Scene. The tool is available as a GUI plugin. To open this plugin, first launch
```
Comment (nkoenig) [inline tutorials/video_recorder.md:6]:
```suggestion
scene. The recorder tool is available as a GUI plugin. To open this plugin, first launch
```

File: tutorials/video_recorder.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+In this plugin, you should see a single button with a video recorder icon.
+Clicking on the button gives you the video format options that are available.
+
+@image html files/video_recorder/video_recorder.png
+
+Once an option is selected, recording starts immediately as indicated by
+a flashing video recorder icon. At anytime that you wish to stop recording,
+click on the flashing icon and select `Stop`. A file dialog window should pop up
+and let you select the path to save the recorded video in. Save the file and
+make sure to also include the file extension in the filename.
```
Comment (nkoenig) [inline tutorials/video_recorder.md:25]: Manually specifying the file extension is not required, right? I think John fixed this.

File: tutorials/video_recorder.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+Playback the video you just saved and you should notice that the resolution
+of the video is based on the size of your 3D Scene window. So if you wish
```
Comment (nkoenig) [inline tutorials/video_recorder.md:28]:
```suggestion
of the video is based on the size of your 3D scene window. So if you wish
```

File: tutorials/video_recorder.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+## Video recorder configurations
+
+A few video recorder parameters can be specified using GUI configurations, see
+the [GUI Configuration](gui_config.html) tutorial for more information.
+If you launched Ignition Gazebo with the
+`video_record_dbl_pendulum.sdf` demo world, the GUI configurations are embedded
+in the world SDF file so you will need to download a copy of the
+[sdf file](https://raw.githubusercontent.com/ignitionrobotics/ign-gazebo/ign-gazebo3/examples/worlds/video_record_dbl_pendulum.sdf).
+and modify the GUI configuration in that file. On the other hand, if launched
```
Comment (nkoenig) [inline tutorials/video_recorder.md:43]:
```suggestion
and modify the GUI configuration in that file. On the other hand, if you launched
```

**Review by nkoenig — CHANGES_REQUESTED** [review]

File: tutorials/video_recorder.md (hunk 4 above)
Comment (iche033) [reply tutorials/video_recorder.md]: bdee780300291a93a6ccc239e65640ca3201276f

File: tutorials/video_recorder.md (hunk 5 above)
Comment (iche033) [reply tutorials/video_recorder.md]: yep just verified that it works without specifying the extension. bdee780300291a93a6ccc239e65640ca3201276f

File: tutorials/video_recorder.md (hunk 6 above)
Comment (iche033) [reply tutorials/video_recorder.md]: bdee780300291a93a6ccc239e65640ca3201276f

File: tutorials/video_recorder.md (hunk 7 above)
Comment (iche033) [reply tutorials/video_recorder.md]: bdee780300291a93a6ccc239e65640ca3201276f

**Review by nkoenig — APPROVED** [review]

**Pull request merged**